src/QueryBuilder/QueryBuilder/QueryBuilderModal/SelectionContainer/SelectionContainer.js

```diff
@@ -0,0 +1,74 @@
+import PropTypes from 'prop-types';
+import { useQuery } from 'react-query';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+
+export const SelectionContainer = (
+  {
+    Selection,
+    availableValues,
+    nameOfComponent,
+    onChange,
+    getParamsSource,
+    source,
+    ...rest
+  },
+) => {
+  const intl = useIntl();
+  const [searchValue, setSearchValue] = useState('');
+  const getSelectOptionsWithPlaceholder = (options) => {
+    if (nameOfComponent === 'Select' && Array.isArray(options)) {
+      return [
+        { value: '', label: intl.formatMessage({ id: 'ui-plugin-query-builder.control.value.placeholder' }), disabled: true },
+        ...options,
+      ];
+    } else return options;
+  };
+
+  const { data } = useQuery({
+    queryKey: [source?.source?.entityTypeId],
+    queryFn: () => getParamsSource(
+      { entityTypeId: source?.source.entityTypeId,
+        source: source?.source.columnName,
+        search: searchValue },
+    ),
+    enabled: source && !availableValues,
+  });
+
+  const filterOptions = (filterText, list) => {
+    // escape special characters in filter text, so they won't be interpreted by RegExp
+    const escapedFilterText = filterText?.replace(/[#-.]|[[-^]|[?|{}]/g, '\\$&');
+
+    setSearchValue(filterText);
+
+    const filterRegExp = new RegExp(`${escapedFilterText}`, 'i');
+    const renderedItems = filterText ? list?.filter(item => item.label.search(filterRegExp) !== -1)
+      :
+      list;
+    const exactMatch = filterText ? (renderedItems.filter(item => item.label === filterText).length === 1) : false;
+
+    return { renderedItems, exactMatch };
+  };
+
+  if (!data && !availableValues) {
+    return null;
+  } else {
+    return (
+      <Selection
+        {...rest}
+        onChange={onChange}
+        filter={filterOptions}
+        dataOptions={getSelectOptionsWithPlaceholder(availableValues) || getSelectOptionsWithPlaceholder(data?.values)}
```

Comment:
please move it to separate variable and then use this value ibside if statement on line 53